code/game/machinery/autolathe.dm

```diff
@@ -228,15 +244,16 @@
 	else
 		for(var/i=1, i<=multiplier, i++)
 			var/obj/item/new_item = new being_built.build_path(A)
-			new_item.autolathe_crafted(src)
 
-			if(length(picked_materials))
-				new_item.set_custom_materials(picked_materials, 1 / multiplier) //Ensure we get the non multiplied amount
-				for(var/x in picked_materials)
-					var/datum/material/M = x
-					if(!istype(M, /datum/material/glass) && !istype(M, /datum/material/iron))
-						user.client.give_award(/datum/award/achievement/misc/getting_an_upgrade, user)
+			if(!(new_item.material_flags & MATERIAL_NO_EFFECTS))
+				new_item.set_custom_materials(materials_used) // the object has effects that change the object depending on the material.
+				if(!isnull(user))
+					// Get an achievement for making a gold toolbox!
+					user.client.give_award(/datum/award/achievement/misc/getting_an_upgrade, user)
```

Comment:
you get an achievement for making any toolbox like this.

Reply:
Well yea.  It was like that originally for some reason.

Reply:
But it checks if it was not made out of metal or glass in the previous call.  Again, not saying its not a stupid achievement, just saying it still tests the same thing.